### ISSUE TITLE
Codecov reporting wrong coverage information.

### ISSUE THREAD

**Issue by lk-geimfari** [lk-geimfari/mimesis#1489]
I have no idea why, but commit 79f6257 (`Add pytest entry in pyproject.toml`) decreased coverage by [42.21%](https://app.codecov.io/gh/lk-geimfari/mimesis/commit/79f625782f7cf0ff1943635abe9cfdc9961ae3f3/indirect-changes). Clearly, Codecov is reporting incorrect coverage information and the question is "Why"?

**Comment by lk-geimfari** [comment]
@sobolevn Would you have any guesses about that? Ever run into anything like this?

**Comment by lk-geimfari** [comment]
This is what happens when we comment out the pytest plugin: https://app.codecov.io/gh/lk-geimfari/mimesis/pull/1502

The coverage increased to 99%.

**Comment by sobolevn** [comment]
Try to move imports inside the fixture functions. `sys.modules` is populated to early.

**Comment by lk-geimfari** [comment]
@sobolevn Do you mean like this?

```python
@pytest.fixture(scope="session")
def _mimesis_cache() -> _CacheCallable:
    from mimesis.locales import Locale
    from mimesis.schema import Field

    cached_instances: dict[Locale, Field] = {}

    def factory(locale: Locale) -> Field:
        if locale not in cached_instances:
            cached_instances[locale] = Field(locale)
        return cached_instances[locale]

    return factory


@pytest.fixture()
def mimesis_locale():  # type: ignore
    """Specifies which locale to use."""
    from mimesis.locales import Locale

    return Locale.DEFAULT
```

**Comment by sobolevn** [comment]
yes

**Comment by lk-geimfari** [comment]
Alas, [didn't ](https://app.codecov.io/gh/lk-geimfari/mimesis/commit/23477c711942f6054dc0b77e1b317663d5384086) helped.

**Comment by mheguy-flo** [comment]
This is due to how the import system works and how you're using `__init__.py`.
Neither are wrong or bad.

When python tries to import mimesis.plugins.pytest, it loads mimesis.__init__.py and looks for plugins.
It doesn't find it, so it looks for the subpackage plugins and loads that (again with __init__.py).

You can see this in action by doing the following: 
Modify the init files in mimesis and plugins by adding this line at the top: `from pathlib import Path; (Path(__file__).parent / "bloop").touch()`
Now create a new python file in the root of the repo.
In it, add this line: `import mimesis.plugins.zoop`
Now run the file from the terminal: `python newfile.py`

You should see 2 new files, both called 'bloop', demonstrating that the import file was interpreted.

----

In terms of a solution, I am less helpful because I don't know the pytest plugin space.

**Comment by lk-geimfari** [comment]
> This is due to how the import system works and how you're using `__init__.py`. Neither are wrong or bad.
> 
> When python tries to import `mimesis.plugins.pytest`, it loads `mimesis.__init__.py` and looks for `plugins`. Because the init refers to most of the files in the package, they're all loaded too early.
> 
> You can see this in action by doing the following: Modify as many files as you dare in the mimesis folder with the following line: `from pathlib import Path; (Path(__file__).parent / Path(__file__).stem).touch()`
> 
> Now run the python repl and enter this: `from mimesis.plugins import bloop`
> 
> You'll get a few new files created for your trouble: ![image](https://private-user-images.githubusercontent.com/146769446/315816286-df635dd0-9e47-4e79-aebb-efdff62c708b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEyNDUxNTcsIm5iZiI6MTcxMTI0NDg1NywicGF0aCI6Ii8xNDY3Njk0NDYvMzE1ODE2Mjg2LWRmNjM1ZGQwLTllNDctNGU3OS1hZWJiLWVmZGZmNjJjNzA4Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyNFQwMTQ3MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YjgwNzRmYWYxN2E5YmM2YjM5NTU2MDJkOTYyMjIzMjViM2MwMzQ2MTY2OGQxOWNiYjhiNDQ3MmEzNTVhNDJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.v8i1ox67wmJBcDcT_xIKlmi4ARiC0cCHYL9-ORpaeyo)

Yeah, I get it, but i did not figured out how can I fix this without breaking anything. Ideas or PR are welcomen.
